### PR TITLE
BUG Fix for Add numeric_only to function signature of DataFrameGroupBy.cumprod and `DataFrameGroupBy.cumsum 

### LINKED ISSUE #53071: BUG: Add ``numeric_only`` to function signature of ``DataFrameGroupBy.cumprod`` and ``DataFrameGroupBy.cumsum`
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

Both methods support numeric only but it goes through args or kwargs, which is annoying signature wise.

### Issue Description

Add the keyword with the appropriate default

### Expected Behavior

See above

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by maushumee**
- [x] closes #53071  
- [x] Tests added and passed
- [x] All code checks passed.
- [x] Added type annotations to new arguments.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by maushumee** [comment]
Hello! I have made changes to include 'numeric_only' flag in the function parameters as described in the issue description scope. Please let me know if we need to also have 'skipna' in the function parameters and remove *args and **kwargs entirely. Thank you!

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -4681,12 +4681,14 @@ def rank(
...
     @Substitution(see_also=_common_see_also)
-    def cumprod(self, *args, **kwargs) -> NDFrameT:
+    def cumprod(self, numeric_only: bool = False, *args, **kwargs) -> NDFrameT:
         """
         Cumulative product for each group.
 
         Parameters
         ----------
+        numeric_only : bool
+            Optional argument with default as False.
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:4691]:
```suggestion
        numeric_only : bool, default False
            Include only float, int, boolean columns.
```

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -4734,18 +4736,20 @@ def cumprod(self, *args, **kwargs) -> NDFrameT:
...
     @Substitution(see_also=_common_see_also)
-    def cumsum(self, *args, **kwargs) -> NDFrameT:
+    def cumsum(self, numeric_only: bool = False, *args, **kwargs) -> NDFrameT:
         """
         Cumulative sum for each group.
 
         Parameters
         ----------
+        numeric_only : bool
+            Optional argument with default as False.
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:4752]:
```suggestion
        numeric_only : bool, default False
            Include only float, int, boolean columns.
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @maushumee

**Pull request merged**